core/node/state_keeper/src/io/mod.rs

```diff
@@ -136,7 +136,7 @@ pub trait StateKeeperIO: 'static + Send + Sync + fmt::Debug + IoSealCriteria {
     /// Marks the transaction as "not executed", so it can be retrieved from the IO again.
     async fn rollback(&mut self, tx: Transaction) -> anyhow::Result<()>;
     /// Marks the transaction as "rejected", e.g. one that is not correct and can't be executed.
-    async fn reject(&mut self, tx: &Transaction, error: &str) -> anyhow::Result<()>;
+    async fn reject(&mut self, tx: &Transaction, error: UnexecutableReason) -> anyhow::Result<()>;
```

Comment:
Bikeshedding: I'd rename the argument to `reason: UnexecutableReason`, so that its name is more expected.